### PR TITLE
Allow to customise chip press elevation

### PR DESCRIPTION
**Pull request by lukaspili**
This PR adds a `pressElevation` property for tappable and selectable chips, rather than applying a constant 8.0 value.

Seems to be consistent with how other material widgets are designed (card, bar, FAB, etc): provide a default elevation matching the Material specs but allow it to be changed/removed. Besides the Material documentation does not seem to state a default value for chip press elevation (or I didn't find it).

The chip widget is very customisable except for this attribute. Adding `pressElevation` should allow the widget to be used in both hard lined Material UIs and some more flat Cupertino UIs where elevation/shadow is not wanted.

Not sure if a test is required for the new attribute?

I also removed a static property `pressedShadowTween` which is not used anywhere.

### CODE REVIEW

**Comment by Hixie** [comment]
Tests are definitely required. :-)

You will want to check obvious things, like whether the attribute works on each of the chip types you added it to, but you also want to check less obvious things like in particular that _not_ setting the elevation, e.g. on things like ActionChip, still results in an elevation when it's pressed.

You may wish to consider using our new golden test technology to test these.
https://github.com/flutter/flutter/wiki/Writing-a-golden-file-test-for-package:flutter

**Comment by lukaspili** [comment]
Ok :) Will do by the end of next week.

**Comment by Hixie** [comment]
@lukaspili any luck with tests?

cc @gspencergoog for review

File: packages/flutter/lib/src/material/chip.dart [hunk 1]
```diff
@@ -262,6 +261,12 @@ abstract class SelectableChipAttributes {
   /// ```
   ValueChanged<bool> get onSelected;
 
+  /// Elevation to be applied on the chip during the press motion.
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/chip.dart]: I'd say something like:
The z-coordinate (elevation) at which to place this material when the press gesture is active.
(and at the other site).

File: packages/flutter/lib/src/material/chip.dart [hunk 2]
```diff
@@ -1153,10 +1183,6 @@ class _RawChipState extends State<RawChip> with TickerProviderStateMixin<RawChip
   Animation<double> enableAnimation;
   Animation<double> selectionFade;
 
-  static final Tween<double> pressedShadowTween = new Tween<double>(
```
Comment (gspencergoog) [inline packages/flutter/lib/src/material/chip.dart]: Thanks for removing the cruft!

**Review by gspencergoog — CHANGES_REQUESTED** [review]
Thanks for the PR, @lukaspili, as @Hixie said above, we just need some tests for this to complete it.  If you don't have time, will work on them eventually, but it'll probably be a while before we get them in. Let us know if you need any help from us.

**Comment by Hixie** [comment]
@lukaspili Will you have the opportunity to integrate the above review comments? It's totally fine if you don't have time to do this, we'd be happy to (eventually) take care of it. Thanks for your contribution!

**Pull request closed without merging**